src/logic/map_objects/tribes/soldier.cc

```diff
@@ -1231,9 +1231,6 @@ void Soldier::defense_update(Game& game, State& state) {
 	if (soldiers.empty() || (get_current_health() < get_retreat_health())) {
 		if (get_retreat_health() > get_current_health()) {
 			assert(state.ivar1 & CF_RETREAT_WHEN_INJURED);
-		}
-
-		if (get_current_health() < get_retreat_health()) {
```

Comment:
Swap the molog calls below - the comparator is the other way around

Reply:
They are the same: the comparator is the other way around, but the operands are also switched. Both conditions match the one in line 1231, so I changed it to this notation for easier understanding. The `else` part corresponds to `soldiers.empty()`.
Thanks for the fast review :)